packages/ogmios/src/Provider/TxSubmitProvider/OgmiosTxSubmitProvider.ts

```diff
@@ -28,16 +32,22 @@ export class OgmiosTxSubmitProvider extends RunnableModule implements TxSubmitPr
   #txSubmissionClient: TxSubmissionClient;
   #logger: Logger;
   #connectionConfig: ConnectionConfig;
+  #handleProvider?: KoraLabsHandleProvider;
 
   /**
    * @param {ConnectionConfig} connectionConfig Ogmios connection configuration
    * @param {Logger} logger object implementing the Logger abstract class
    * @throws {TxSubmission.errors}
    */
-  constructor(connectionConfig: ConnectionConfig, { logger }: ProviderDependencies) {
+  constructor(
+    connectionConfig: ConnectionConfig,
+    { logger }: ProviderDependencies,
+    handleProvider?: KoraLabsHandleProvider
```

Comment:
Is there a reason to depend on specific implementation here?
```suggestion
    handleProvider?: HandleProvider
```

Reply:
af17ac42b57672c085e8d89d7f2565a979aaad0c should fix this